src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,398 @@
+        "$.channels.*.parameters.*"
+      ],
+      "then": {
+        "field": "description",
+        "function": "truthy"
+      }
+    },
+    "asyncapi2-payload-default": {
+      "description": "Default must be valid against its defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload.default^",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload.default^",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload.default^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "default",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-payload-examples": {
+      "description": "Examples must be valid against their defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload.examples^",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload.examples^",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload.examples^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "$.examples.*",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-payload-unsupported-schemaFormat": {
+      "description": "Message schema validation is only supported with default unspecified `schemaFormat`.",
+      "severity": 2,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits.*",
+        "$.components.messages.*",
+        "$.channels.*.[publish,subscribe].message"
+      ],
+      "then": {
+        "field": "schemaFormat",
+        "function": "undefined"
+      }
+    },
+    "asyncapi2-payload": {
+      "description": "Payloads must be valid against AsyncAPI Schema object.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload"
+      ],
+      "then": {
+        "function": "asyncApi2PayloadValidation"
+      }
+    },
+    "asyncapi2-schema-default": {
+      "description": "Default must be valid against its defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.schemas.*.default^",
```

Comment:
Is the ^ a JSON Path feature I'm not familiar with?

Reply:
@philsturgeon It's part of what jsonpath-plus brings:
- `^` to return the parent.
- `*~` to return the keys